sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ReplaceNullWithFalseInPredicateSuite.scala

```diff
@@ -456,15 +456,21 @@ class ReplaceNullWithFalseInPredicateSuite extends PlanTest {
 
   private def testMerge(originalCond: Expression, expectedCond: Expression): Unit = {
     val func = (rel: LogicalPlan, expr: Expression) => {
+      val target = rel.as("target")
+      val source = rel.as("source")
       val assignments = Seq(
-        Assignment('i, 'i),
-        Assignment('b, 'b),
-        Assignment('a, 'a),
-        Assignment('m, 'm)
+        Assignment(UnresolvedAttribute("i"), UnresolvedAttribute("source.i")),
+        Assignment(UnresolvedAttribute("b"), UnresolvedAttribute("source.b")),
+        Assignment(UnresolvedAttribute("a"), UnresolvedAttribute("source.a")),
+        Assignment(UnresolvedAttribute("m"), UnresolvedAttribute("source.m"))
```

Comment:
A question related to the above discussion, all of these work before and after this PR? Or, is this different test coverage?

> For UPDATE SET a = source.a, I think it works before and after this fix. @Ngone51 can you confirm?
> Yes, it works before and after this fix.

Reply:
Yes, it works both before and after. (Not different test coverage). But these changes are required after we introduced `dedup` to `MergeIntoTable` in this PR. Otherwise, tests can fail with the ambiguous reference error.

Reply:
Actually we don't have to test self-merge here. Can we create a new relation for testing merge? so that we don't need to rewrite the conditions.

Reply:
OK.